tests/model/test_model.py

```diff
@@ -783,6 +783,36 @@ def test__handle_message_event(self, mocker, user_profile, response,
         # LOG REMAINS THE SAME IF UPDATE IS FALSE
         assert model.msg_list.log == log
 
+    @pytest.mark.parametrize('found_newest', [
+        (True),
+        pytest.param(False,
+                     marks=pytest.mark.xfail(reason="Unread count bug")),
+        ])
+    @pytest.mark.parametrize('event', [
+        ({'message': {'type': 'stream', 'stream_id': 1, 'subject': 'HELLO',
+          'id': 1}, 'flags': []}),
+        ])
+    def test__handle_message_event_set_count_all_narrows(self, mocker, event,
+                                                         model, found_newest,
```

Comment:
This uses `model`, which has an initial default narrow of 'all messages', so this doesn't replicate the 'in a different narrow' test case we started with. This didn't turn out to be the issue, so I'm not sure if we want to explore behavior over different narrows.

Reply:
Being in a different narrow won't make a difference since it's only `found_newest` which affects behaviour?

Reply:
Renamed the test to `test__handle_message_event_set_count_found_newest`, since we are not directly considering different narrows.

PS. Now that I look at this, it feels a little too much to write such a big test for such a small change.
Wouldn't a comment saying "Update counts irrespective of found_newest" have been enough?